### PR TITLE
Fix errors pointed by shellcheck

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: tests/upgrade/test_upgrade_from.sh [hunk 1]
```diff
@@ -35,75 +35,67 @@ LOCAL_FORTIO_LOG=/tmp/fortio_local.log
 POD_FORTIO_LOG=/tmp/fortio_pod.log
 
 # Make sure to change templates/*.yaml with the correct address if this changes.
-TEST_NAMESPACE=test
+TEST_NAMESPACE="test"
```
Comment (baodongli) [inline tests/upgrade/test_upgrade_from.sh]: Quotes are not needed. I saw that a lot of changes simply add double quotes, which are not necessary at all.

File: tests/upgrade/test_upgrade_from.sh [hunk 2]
```diff
@@ -35,75 +35,67 @@ LOCAL_FORTIO_LOG=/tmp/fortio_local.log
...
-   kubectl apply -n istio-system -f ${ISTIO_ROOT}/istio.yaml
+   kubectl apply -n istio-system -f "${ISTIO_ROOT}/istio.yaml"
 }
 
 gitChangeToTag() {
-   original_branch=`git rev-parse --abbrev-ref HEAD`
-   git checkout tags/${1} || die "Cannot git checkout, ensure your workspace is clean."
-   cur_branch=`git rev-parse --abbrev-ref HEAD`
-   cur_sha=`git rev-parse HEAD`
+   original_branch=$(git rev-parse --abbrev-ref HEAD)
```
Comment (baodongli) [inline tests/upgrade/test_upgrade_from.sh]: has the original branch been set at line 40?

File: tests/upgrade/test_upgrade_from.sh (hunk 1 above)
Comment (jwendell) [reply tests/upgrade/test_upgrade_from.sh]: ShellCheck complains about lack of quotes

File: tests/upgrade/test_upgrade_from.sh (hunk 2 above)
Comment (jwendell) [reply tests/upgrade/test_upgrade_from.sh]: yep, I didn't pay much attention to the logic of this script, I believe this line is not necessary. Want me to remove it?

File: tests/upgrade/test_upgrade_from.sh [hunk 3]
```diff
@@ -35,75 +35,67 @@ LOCAL_FORTIO_LOG=/tmp/fortio_local.log
...
-    cur_time=$(date +%s)
     writeMsg "Restarting deployment ${1}"
-    kubectl patch deployment ${1} -n ${TEST_NAMESPACE} -p'{"spec":{"template":{"spec":{"containers":[{"name":"echosrv","env":[{"name":"RESTART_","value":"1"}]}]}}}}'
+    kubectl patch deployment "${1}" -n "${TEST_NAMESPACE}" -p'{"spec":{"template":{"spec":{"containers":[{"name":"echosrv","env":[{"name":"RESTART_","value":"1"}]}]}}}}'
 }
 
 writeMsg() {
-    printf "\n\n****************\n\n${1}\n\n****************\n\n"
-
+    printf "\\n\\n****************\\n\\n%s\\n\\n****************\\n\\n" "${1}"
```
Comment (baodongli) [inline tests/upgrade/test_upgrade_from.sh]: change ${1} to ${@}
No quotes are needed when calling writeMsg()
for example:
       writeMsg no quotes is needed

File: tests/upgrade/test_upgrade_from.sh (hunk 1 above)
Comment (baodongli) [reply tests/upgrade/test_upgrade_from.sh]: surprised that it'd complain about lacking of quotes in the above statement.

File: tests/upgrade/test_upgrade_from.sh (hunk 3 above)
Comment (jwendell) [reply tests/upgrade/test_upgrade_from.sh]: It seems it doesn't work. It adds line breaks to printf.

Comment (baodongli) [reply tests/upgrade/test_upgrade_from.sh]: it'll work with echo. 

But I think it's fine as is

File: tests/upgrade/test_upgrade_from.sh (hunk 2 above)
Comment (baodongli) [reply tests/upgrade/test_upgrade_from.sh]: it would be good since you also removed some lines

**Comment by jwendell** [comment]
@baodongli done.

**Pull request merged**